// Copyright 2010-2014 Google
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.


syntax = "proto3";

package operations_research;

// Storage for IntVars.
message IntVarAssignment {
  string var_id = 1;
  int64 min = 2;
  int64 max = 3;
  bool active = 4;
}

// Storage for IntervalVars.
message IntervalVarAssignment {
  string var_id = 1;
  int64 start_min = 2;
  int64 start_max = 3;
  int64 duration_min = 4;
  int64 duration_max = 5;
  int64 end_min = 6;
  int64 end_max = 7;
  int64 performed_min = 8;
  int64 performed_max = 9;
  bool active = 10;
}

// Storage for SequenceVars.
message SequenceVarAssignment {
  string var_id = 1;
  repeated int32 forward_sequence = 2;
  repeated int32 backward_sequence = 3;
  repeated int32 unperformed = 4;
  bool active = 5;
}

// This message indicates how the assignment was produced.
message WorkerInfo {
  int32 worker_id = 1;
  string bns = 2;
}

// Global container for all assignment variables and objective
message AssignmentProto {
  repeated IntVarAssignment int_var_assignment = 1;
  repeated IntervalVarAssignment interval_var_assignment = 2;
  repeated SequenceVarAssignment sequence_var_assignment = 6;
  IntVarAssignment objective = 3;
  WorkerInfo worker_info = 4;
  bool is_valid = 5;
}

